.github/workflows/ci-master.yml

```diff
@@ -27,11 +27,11 @@ jobs:
       - name: Install npm and dependencies
         run: npm install -g npm@latest && npm install
 
+      - name: Lint
+        run: npm run lint
+
       - name: Build
         run: npm run build --if-present
 
       - name: Test
         run: npm run test
-
-      - name: Lint
-        run: npm run lint
```

Comment:
on master often the slowest is `lint`, and the fastest is `test`.
should we change this to `test` -> `build` -> `lint` ?

Reply:
`test` requires `build` first, else we trigger [this issue](https://github.com/n8n-io/n8n/actions/runs/3158050095/jobs/5139642049). If we count `build+test` together, then `lint` is faster.

Or we could also do a partial `build` of only the packages needed by tests so that `test` can run first, and then `lint` and full `build` (or these two swapped around, mostly identical), but this option is prolly overkill for the gain.

Reply:
`ts-jest` is supposed to transpile dependencies. so `test` requiring `build` sounds like a bug to me.

Reply:
Unbuilt `n8n-core` cannot be found by `ts-jest` because it is symlinked from `node_modules/n8n-core` to `packages/core` because of workspaces. Do we want to avoid this symlinking to prevent this coupling?

Also I wonder if we could explore having `ts-jest` use the output of `build` to avoid having transpilation happen twice. Maybe as a separate PR.

Reply:
that sounds like we need to improve out build tooling then.
let's merge this for now, and revisit further improvements in other PRs.